core/src/main/resources/hudson/slaves/SlaveComputer/slave-agent.jnlp.jelly

```diff
@@ -89,6 +89,21 @@ THE SOFTWARE.
 
         <argument>-url</argument>
         <argument>${rootURL}</argument>
+
+        <j:if test="${rootURL!=app.rootUrlFromRequest}">
```

Comment:
Untested but

```suggestion
        <j:if test="${!launcher.webSocket and rootURL!=app.rootUrlFromRequest}">
```

would preserve the fix for WebSocket agents while reverting to the original (IMO dubious) behavior for TCP agents.